### ISSUE TITLE
QT with GO without docker

### ISSUE THREAD

**Issue by ManPython** [therecipe/qt#565]
Is this possible in some short way to compile QT with GO without docker?
Anybody know working solution?
There is many reasons that docker is not super, eg. in case that docker can't be instaled on machine.

**Comment by amlwwalker** [comment]
Well.... you could look inside the docker file and see what they are doing to configure the docker container, and then replicate those steps on your machine. You will have to find any Windows equivalents to whatever they are doing because it will be a flavour of linux in there. This is exactly why docker exists, because if you start out to do this... you will... i suspect.... go through pain....

Can you install a virtual machine on your Windows computer (Virtual Box + Ubuntu for instance) - you could then install docker in the Ubuntu image, share a directory with the Host Windows machine and voila...

**Comment by ManPython** [comment]
It's not the 'voila' : )  ..in case that there is many isues around docker on windows.
It is Star Trek GO solution like in way to going to the cosmos.. check this: "install a virtual machine on your Windows computer (Virtual Box + Ubuntu for instance) - you could then install docker in the Ubuntu image, share a directory with the Host Windows"..

This is realy funny coz.. similary looks in other project
https://github.com/visualfc/goqt/blob/master/doc/install.md

And comparing this to the Python where is just 'pip instal pyqt' as grab small package and use.. GO and GUI it's nightmare.

Sooo this is real pain.
And.. next with the project: google it - "docker working very slow" ..compilation is pain..

**Comment by therecipe** [comment]
Hey

Yeah, sorry the installation situation is a little bit confusing at the moment.
I will cleanup the wiki and make it more obvious that you don't need to use docker at all to use the binding (it's just one of the many ways to use it).

Of course you can simply install Qt and then just run the setup, maybe take a look here: https://github.com/therecipe/qt/wiki/Installation-on-Windows#installing-qt

**Comment by ManPython** [comment]
I failed in both: MSYS2 and docker - both solutions are redundant in my opinion. 

I appreciate the work on the project, but in the general work of using the solution, additional things only complicate the work. More if need additional configurations that are not defauld but required. This is confussing.

The idea of GO was result of consensus from python philosophy around packages (go get / pip install) as easy way. And we have the main GUI that blockig growing the GO and the using therecipe/qt.

@therecipe 
I hopte that the project can be optimalised here.

**Comment by therecipe** [comment]
@ManPython
Yeah, I know that the situation is no perfect at the moment.
But the docker and MSYS2 solutions have there purpose as well.
MSYS2 for example allows static linking on windows and the docker stuff is mainly used for CI/CD.

I think that I just need to simplify the wiki to make things more obvious.
But at the moment you should be able to get the package working on windows/mac/linux, by simply downloading and installation Qt + Go, then running `go get ...` and finally running `qtsetup`.

**Comment by ManPython** [comment]
@therecipe 
Ok, It be great if You can remeber about me and this tread and call me in any changes on wiki or other as philosophy with using QT and GO.

On this moment some situation about QT is often problem with SSD drives where QT instalation need 1GB and puting on drive about 10GB. If someone is not familar with this all.. You know.. for python PyQt is around 130MB.. and idea to use GO was that the compilation to the .exe is bether than Python (slow execution of bigest apps).

Thx for attention.

**Comment by therecipe** [comment]
@ManPython 
Yeah, the main difference between the installation process of pyqt and this binding is that this binding is bootstrapping itself depending on the current enviorment whether pyqt simply ships pre-build libraries of their code + a stripped down version of Qt.
However, I will try to find a way to make the installation of this binding as "pip install pyqt" in the future.

And I will call you once I worked on the wiki.

**Comment by ManPython** [comment]
To the social information
http://blog.qt.io/blog/2018/06/13/qt-python-5-11-released/#comment-1204043

> Although we are using Go for some of our own infrastructure, at this point in time there are no plans to add bindings for the language. On the positive side, there are quite a few existing language bindings for Qt already:
